sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java

```diff
@@ -359,12 +361,70 @@ private static DefaultResourceSet from(
             resourceSetBuilder.memory(Double.valueOf(memory));
         }
 
-        if (rawEndpoints != null) {
-            resourceSetBuilder.addPorts(rawEndpoints);
+        if (rawPorts != null) {
+            resourceSetBuilder.addResource(from(role, principal, rawPorts));
         }
 
         return resourceSetBuilder
                 .id(id)
                 .build();
     }
+
+    private static ResourceSpec from(String role, String principal, WriteOnceLinkedHashMap<String, RawPort> rawPorts) {
+        Collection<PortSpec> portSpecs = new ArrayList<>();
+        Protos.Value.Builder portsValueBuilder = Protos.Value.newBuilder().setType(Protos.Value.Type.RANGES);
+        String envKey = null;
+        for (Map.Entry<String, RawPort> portEntry : rawPorts.entrySet()) {
+            String name = portEntry.getKey();
+            RawPort rawPort = portEntry.getValue();
+            Protos.Value.Builder portValueBuilder = Protos.Value.newBuilder()
+                    .setType(Protos.Value.Type.RANGES);
+            portValueBuilder.getRangesBuilder().addRangeBuilder()
+                    .setBegin(rawPort.getPort())
+                    .setEnd(rawPort.getPort());
+            portsValueBuilder.mergeRanges(portValueBuilder.getRanges());
+            if (envKey == null) {
+                envKey = rawPort.getEnvKey();
+            }
+
+            if (rawPort.getVip() != null) {
+                final RawVip rawVip = rawPort.getVip();
+                final String protocol =
+                        StringUtils.isEmpty(rawVip.getProtocol()) ? DEFAULT_VIP_PROTOCOL : rawVip.getProtocol();
+                final String vipName = StringUtils.isEmpty(rawVip.getPrefix()) ? name : rawVip.getPrefix();
+                portSpecs.add(new NamedVIPSpec(
+                        Constants.PORTS_RESOURCE_TYPE,
+                        portValueBuilder.build(),
+                        role,
+                        principal,
+                        rawPort.getEnvKey(),
+                        name,
+                        protocol,
+                        toVisibility(rawVip.isAdvertised()),
```

Comment:
Is the `advertised` YAML field documented?  It doesn't seem to be documented in the dev guide.  What does it mean to "advertise" a VIP?